src/libtiled/wangset.cpp

```diff
@@ -470,6 +521,42 @@ bool WangSet::wangIdIsUsed(WangId wangId) const
     return mWangIdToWangTile.contains(wangId);
 }
 
+bool WangSet::wildWangIdIsUsed(WangId wangId) const
+{
+    if (isEmpty())
+        return false;
+    if (!wangId)
+        return true;
+
+    auto i = wangId.variationsBegin(mEdgeColors, mCornerColors);
+    auto end = wangId.variationsEnd(mEdgeColors, mCornerColors);
+
+    while (true) {
+        if (wangIdIsUsed(*i))
+            return true;
+
+        if (++i == end) {
+            if (wangIdIsUsed(*i))
+                return true;
+            break;
+        }
+    }
+
+    return false;
+}
+
+bool WangSet::isComplete() const
+{
+    if (isEmpty())
+        return false;
+    return (unsigned) mWangIdToWangTile.uniqueKeys().size() == completeSetSize();
```

Comment:
I also realized since it should be possible to assign a wangId with 0s (wildcards) this has to check for the count of unique keys with no wild cards. So I'll write a specific function for this in WangSet based on your suggestion and that.